### ISSUE TITLE
Model tuning

### ISSUE THREAD

**Issue by millerh1** [Bishop-Laboratory/autoPLIER-analysis#27]
New subtasks:

- [x] Need to figure out whether the sparsity is being measured correctly
- [x] Need to test the sparsity of the U matrix
- [x] Need to scan a smaller range of regularization params
- [x] Need a sparsity vs performance plot
- [x] Need to come up with a "sweet spot" estimate. Need most interpretable model within the cross-over range of sparsity vs performance. 
- [x] Can we come up with a method for setting the number of LVs? Maybe number of PCA components that explain X percent of variance -- this gives the range for LV number (future direction)
- [x] Sweep through various numbers of LBs: 100, 200, ... For each, compute cell type classification performance -- what % variance would be explained by that number of PCs?

@Julian-Borbeck

**Comment by millerh1** [comment]
Tried "Can we come up with a method for setting the number of LVs? Maybe number of PCA components that explain X percent of variance -- this gives the range for LV number (future direction)" but it did not work

**Comment by millerh1** [comment]
Might possibly try the Jackstraw Plot approach from PCA - https://satijalab.org/seurat/reference/jackstrawplot

**Comment by millerh1** [comment]
We could take the JackStraw approach and instead of permuting at the gene level, we could permute at the pathway level and compare against "random" pathways. 

This is the original paper for the JackStraw: https://pubmed.ncbi.nlm.nih.gov/25336500/

**Comment by millerh1** [comment]
New approach:
1. Set LVs and learning rate to a reasonable default
2. Implement a function `learning_rate_plot()` that creates a plot which the user can use to diagnose the learning rate.
3. We also provide a vignette that helps with this. Vignette could say "if you are finding this takes to long, try increasing/decreasing it."

No automation -- sounds like a good thing to talk about in our discussion section!

**Comment by millerh1** [comment]
Still to accomplish:
 - [ ] Pick reasonable default for LVs
 - [ ] Pick reasonable default for learning rate

**Comment by millerh1** [comment]
Chose 100LVs for default and learning rate of 0.00156

**Comment by millerh1** [comment]
Note for @Julian-Borbeck probably need a plot that shows how we arrived at this value for learning rate
